enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,666 @@
+  operate reliably, but may respond more slowly.
+* This enhancement assumes that the configuration of a management CPU
+  pool is done as part of installing the cluster, and cannot be
+  changed later. Future enhancements may address enabling or
+  reconfiguring the feature described here on an existing cluster.
+* This enhancement describes partitioning concepts that could be
+  expanded to be used for other purposes. Use cases for partitioning
+  workloads for other purposes may be addressed by future
+  enhancements.
+
+## Proposal
+
+[Previous attempts](https://github.com/openshift/enhancements/pull/628)
+to solve this problem focused on reducing or scaling requests so that
+the normal scheduling criteria could be used to safely place them on
+CPUs in the shared pool. This proposal reframes the problem, so that
+instead of considering "scaling" or "tuning" the requests we think about
+"isolating" or "partitioning" the workloads away from the non-management
+workloads. This view of the problem is more consistent with how the
+requirement was originally presented by the customer.
+
+We want to define "management workloads" in a flexible way. For the
+purposes of this document, "management workloads" include all
+OpenShift core components necessary to run the cluster, any add-on
+operators necessary to make up the "platform" as defined by telco
+customers, and operators or other components from third-party vendors
+that the customer deems as management rather than operational. It is
+important to note that not all of those components will be delivered
+as part of the OpenShift payload and some may be written by the
+customer or by vendors who are not our partners.
+
+The basic proposal is to provide a way to identify management
+workloads at runtime and to use CRI-O to run them on a user-selected
+set of CPUs, while other workloads will be prevented from running
+there. This effectively gives 3 pools of CPUs
+
+* **management** -- a user-defined cpuset that restricts where
+  management components run
+* **dedicated** -- CPUs used for exclusive workloads
+* **shared** -- CPUs in the set `management + !dedicate`
+
+Because the management CPU pool is restricted to management workloads,
+the shared CPU pool will need other CPUs to support burstable or
+best-effort workloads that are not identified as management workloads.
+
+We want the isolation to be in place from the initial buildout of the
+cluster, to ensure predictable behavior. Therefore, the feature must
+be enabled during installation. To enable the feature, the user will
+specify the set of CPUs to run all management workloads. When the
+management CPU set is not defined, the feature will be completely
+disabled.
+
+We generally want components to opt-in to being considered management
+workloads. Therefore, for a regular pod to be considered to contain a
+management workload it must be labeled with a *workload label*,
+`workload.openshift.io/{pool_name}`. For now, we will focus on
```

Comment:
what is the API support level of these?

I would prefer a very explicit "unsupported" keyword in these labels.

Reply:
`workload.internal.openshfit.io/pool-name`

Reply:
I don't know about "unsupported". The feature will be supported, right? And "internal" implies that it shouldn't be used by components that aren't part of the core release payload, but we must support add-on operators using it.

If the feature is only enabled with a feature gate, is that sufficient to communicate that it is an alpha feature?